azerothcore/azerothcore-wotlk
# feat(Core/Config): Parameter to set all creatures with WP movement active

## CHANGES PROPOSED:
- Introduce a new parameter for the worldserver configuration:
  "SetAllCreaturesWithWaypointMovementActive"
  This will set all creatures active which use waypoint movement after they are loaded (on grid load). I used this for testing but it could be useful if you have enough CPU power and combine this with "PreloadAllNonInstancedMapGrids" which enables all creatures to start their movement on server startup.
- Remove max visibility distance for active creatures, as this is handled by [creature_addon.isLarge](http://www.azerothcore.org/wiki/creature_addon#islarge)

## ISSUES ADDRESSED:
none

## TESTS PERFORMED:
- tested build on Ubuntu 16.04 / clang 7
- tested successfully in-game

## HOW TO TEST THE CHANGES:
Enable the parameter "SetAllCreaturesWithWaypointMovementActive" and visit a few creatures with waypoint movement, e.g.:
- `.go cr id 3237` (Kodos)
- `.go cr id 1738` (Deathguard Terrence)
They should now keep moving around even if no player is near.

## KNOWN ISSUES AND TODO LIST:
none

## Target branch(es):
 
## How to test AzerothCore PRs
 
When a PR is ready to be tested, it will be marked as **[WAITING TO BE TESTED]**.

You can help by testing PRs and writing your feedback here in the PR's page on GitHub. Follow the instructions here:

http://www.azerothcore.org/wiki/How-to-test-a-PR
